draco/analysis/transform.py

```diff
@@ -1265,3 +1257,213 @@ def process_finish(self) -> Union[containers.SiderealStream, containers.RingMap]
         self.mixed_data = None
 
         return data
+
+
+class Downselect(task.SingleTask):
+    """Apply axis selections to a container.
+
+    Apply slice or `np.take` operations across multiple axes of a container.
+    The selections are applied to every dataset.
+
+    If a dataset is distributed, there must be at least one axis not included
+    in the selections.
+
+    Attributes
+    ----------
+    selections : dict
+        Dict formatted as {<ax_name>:<selection>,} for any axes
+        that should be downselected before applying the reduction.
+        Selections can be either slices or a list/tuple of indices.
+    """
+
+    selections = config.Property(proptype=dict)
```

Comment:
I don't think we should reinvent another way of specifying the `selections` in here. Just re-use what's available in `draco.core.io.BaseLoadFiles`. You'll need to find some way of using the `_resolve_sel` method here, cleanest would be moving it all out into a `SelectionsMixin` class that both inherit from, but tbh you could just inherit from `BaseLoadFiles` and ignore all it's other functionality.